src/Cursors.ts

```diff
@@ -137,6 +137,31 @@ export default class Cursors extends EventEmitter<CursorsEventMap> {
     const channel = this.getChannel();
     return await this.cursorHistory.getLastCursorUpdate(channel, this.options.paginationLimit);
   }
+
+  async getSelf(): Promise<CursorUpdate | undefined> {
+    const allCursors = await this.getAll();
+    const self = this.space.getSelf();
+    if (!self || !allCursors) {
+      return undefined;
+    }
+    const cursorUpdates: CursorUpdate[] = Object.values(allCursors).filter(
+      (update): update is CursorUpdate => 'clientId' in update,
+    );
+    return cursorUpdates.find((update) => update.clientId === self.clientId);
```

Comment:
I think because we key the result of `getAll` by `connectionId`, we can change the body of `getSelf` to be just:

```ts
async getSelf(): Promise<CursorUpdate | undefined> {
   const self = this.space.getSelf();
   if (!self) return;

   const allCursors = await this.getAll();
   return allCursors[self.connectionId];
}
```

I think you were mislead by the https://github.com/ably-labs/spaces/blob/main/src/CursorHistory.ts#L10, and the `CursorsLastPostion` type which actually should be removed as it's not used anymore (and then the above will type check too)